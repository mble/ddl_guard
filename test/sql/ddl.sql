SELECT current_setting('ddl_guard.enabled', true) = 'on' AS ddl_guard_enabled;
SELECT current_setting('is_superuser', true) = 'on' AS is_superuser;
CREATE TABLE foobar (id serial PRIMARY KEY);
CREATE FUNCTION foobar_trigger() RETURNS TRIGGER AS $$
BEGIN
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;
