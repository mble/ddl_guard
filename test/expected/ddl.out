SELECT current_setting('ddl_guard.enabled', true) = 'on' AS ddl_guard_enabled;
 ddl_guard_enabled 
-------------------
 t
(1 row)

SELECT current_setting('is_superuser', true) = 'on' AS is_superuser;
 is_superuser 
--------------
 f
(1 row)

CREATE TABLE foobar (id serial PRIMARY KEY);
ERROR:  Non-superusers are not allowed to execute DDL statements
CONTEXT:  PL/pgSQL function ddl_guard_check() line 4 at RAISE
CREATE FUNCTION foobar_trigger() RETURNS TRIGGER AS $$
BEGIN
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;
ERROR:  Non-superusers are not allowed to execute DDL statements
CONTEXT:  PL/pgSQL function ddl_guard_check() line 4 at RAISE
